sdcm/utils/version_utils.py

```diff
@@ -4,65 +4,161 @@
+            The value of the variable must be replaced by one of the supported values (URL_ENV_VARIABLES[BASE_ARCH_KEY])
+        3. "releasever" URL - The URL contains the special variable "$releasever".
+            The value of the variable must be replaced by one of the supported values (
+            URL_ENV_VARIABLES[RELEASE_VERSION_KEY])
+        4. Multiple special variables - The URL contains multiple special variables.
+            Each special variable replaced with supported values.
+    """
 
+    all_scylla_version_urls = set()
+    for url_regex, url_format in url_regex_details:
+        for pattern in scylla_patterns:
+            packages_details = url_regex.match(pattern)
+            if not packages_details:
+                continue
+
+            version_url = url_format.format(**packages_details.groupdict())
+            is_base_arch_found = BASE_ARCH_KEY in version_url
+            is_release_version_found = RELEASE_VERSION_KEY in version_url
+
+            if is_base_arch_found and is_release_version_found:
+                for base_arch_val in URL_ENV_VARIABLES[BASE_ARCH_KEY]:
+                    for release_version in URL_ENV_VARIABLES[RELEASE_VERSION_KEY]:
+                        all_scylla_version_urls.add(version_url.replace(
+                            BASE_ARCH_KEY, base_arch_val).replace(RELEASE_VERSION_KEY, release_version))
+            elif is_base_arch_found:
+                for base_arch_val in URL_ENV_VARIABLES[BASE_ARCH_KEY]:
+                    all_scylla_version_urls.add(version_url.replace(BASE_ARCH_KEY, base_arch_val))
+            elif is_release_version_found:
+                for release_version in URL_ENV_VARIABLES[RELEASE_VERSION_KEY]:
+                    all_scylla_version_urls.add(version_url.replace(RELEASE_VERSION_KEY, release_version))
+            else:
+                all_scylla_version_urls.add(version_url)
+    return list(all_scylla_version_urls)
+
+
+def validate_url_scylla_version_file(scylla_file_version):
+    response = requests.get(url=scylla_file_version)
+    response_status = response.status_code
+    if response_status != 200:
+        raise ValueError(f"The URL '{scylla_file_version}' is invalid (got status code '{response_status}')")
+    if not response.text:
+        raise ValueError(f"The file that downloaded from url '{scylla_file_version}' is empty")
+    return scylla_file_version
+
+
+def extract_all_scylla_urls_from_file(urls):
+    result, all_scylla_version_urls = [], set()
+    if not isinstance(urls, list):
+        urls = [urls]
+
+    for url in urls:
+        all_scylla_version_urls.update(extract_all_urls_from_scylla_patterns(
+            scylla_patterns=get_all_scylla_url_patterns(url=url), url_regex_details=URL_REGEX_AND_FORMAT_LIST))
+
+    for thread in ParallelObject(objects=list(all_scylla_version_urls), timeout=SCYLLA_URL_RESPONSE_TIMEOUT).run(
+            func=validate_url_scylla_version_file, ignore_exceptions=True):
```

Comment:
why `ignore_exceptions=True` ?
ending up with only `VERSION_NOT_FOUND_ERROR` as the error is **all** of those url fail the check, isn't helping much

Reply:
You're right, that's the purpose of the function.
But, the file can contain lots of links and some of them are not Scylla's.
The primary purpose of the function is to make sure that the version of the version is correct or not.
Unfortunately, I have to ignore errors because I get an error for manger links.
If necessary, I can add special treatment for this manger case.